### PR TITLE
#11241: Replace tt_lib in models/demos/bert and falcon7b_common

### PR DESCRIPTION
**Pull request by KalaivaniMCW**
### Ticket
Link to Github Issue #11242 

### Problem description
Replace usage of tt_lib with ttnn

### What's changed
- Replaced usage of tt_lib with ttnn
- Also added device and profiler modules from tt_lib_bindings.cpp to  ttnn.experimental.*

### Checklist

- [x] Post commit CI https://github.com/tenstorrent/tt-metal/actions/runs/10390425575

- [ ] Demo tests SC https://github.com/tenstorrent/tt-metal/actions/runs/10388620729

- [ ] t3k demos https://github.com/tenstorrent/tt-metal/actions/runs/10389230627

- [ ] nightly FD https://github.com/tenstorrent/tt-metal/actions/runs/10388617522

- [ ] PC model tests https://github.com/tenstorrent/tt-metal/actions/runs/10380749509

- [x] model perf https://github.com/tenstorrent/tt-metal/actions/runs/10386170948

- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by djordje-tt — APPROVED** [review]

File: models/demos/falcon7b_common/tests/run_falcon_end_to_end.py [hunk 1]
```diff
@@ -230,12 +230,12 @@ def run_test_FalconCausalLM_end_to_end(
...
             )
         for device in devices:
-            tt_lib.device.Synchronize(device)
+            ttnn.synchronize_device(device)
         profiler.end("first_model_run_with_compile", force_enable=e2e_perf)
 
         # Dump device profiler data before second run to avoid exceeding profiler memory limits when using tracy
         for device in devices:
-            tt_lib.device.DumpDeviceProfiler(device)
+            ttnn.experimental.device.DumpDeviceProfiler(device)
```
Comment (skhorasganiTT) [inline models/demos/falcon7b_common/tests/run_falcon_end_to_end.py:238]: ttnn.experimental.device does not exist as far as I am aware, has this been tested?

File: models/demos/falcon7b_common/tt/falcon_model.py [hunk 2]
```diff
@@ -286,7 +284,7 @@ def forward(
 
             if device_perf_run and idx % 8 == 0:
                 for i in range(self.num_devices):
-                    tt_lib.device.DumpDeviceProfiler(layer_output[i].device())
+                    ttnn.experimental.device.DumpDeviceProfiler(layer_output[i].device())
```
Comment (skhorasganiTT) [inline models/demos/falcon7b_common/tt/falcon_model.py:287]: Same comment as above

**Review by skhorasganiTT — COMMENTED** [review]
Could you run the model perf and model demo pipelines for this change and post the links when passing?

File: models/demos/falcon7b_common/tests/run_falcon_end_to_end.py (hunk 1 above)
Comment (KalaivaniMCW) [reply models/demos/falcon7b_common/tests/run_falcon_end_to_end.py:238]: We changed this based on the tt-eager bindings, 
https://github.com/tenstorrent/tt-metal/blob/802e9a3edd63d5eec5837ae09367a03ff8c830a9/ttnn/cpp/ttnn/deprecated/tt_lib/csrc/tt_lib_bindings.cpp#L210 
and based on the inputs we get from TTNN team in this sheet here
https://docs.google.com/spreadsheets/d/1FKSFDpkM7iXPYQ8nMgjwLkBQK50J6kxFk25j4s-XaZA/edit?usp=sharing

Comment (skhorasganiTT) [reply models/demos/falcon7b_common/tests/run_falcon_end_to_end.py:238]: @KalaivaniMCW were you able to run the model pipelines I suggested above? If they pass then there is no issue. However, I tried `ttnn.experimental.device` earlier today and it did not exist so I would expect it to fail. @arakhmati could you confirm on this?

**Comment by KalaivaniMCW** [comment]
@skhorasganiTT I will rebase and  trigger the model perf and model demo pipelines. yesterday most CI runs we triggered ended up with cancelled jobs. not sure if it's the load or something else. so i'll rebase and let you know.

File: models/demos/falcon7b_common/tests/run_falcon_end_to_end.py (hunk 1 above)
Comment (KalaivaniMCW) [reply models/demos/falcon7b_common/tests/run_falcon_end_to_end.py:238]: @skhorasganiTT
As per @eyonland 's suggestion, I have imported the device module from tt_lib_bindings as `ttnn.experimental.device` 
and ran Model perf CI https://github.com/tenstorrent/tt-metal/actions/runs/10386170948. 
any other specific tests to track these changes?

Comment (skhorasganiTT) [reply models/demos/falcon7b_common/tests/run_falcon_end_to_end.py:238]: Great, thank you, that should be sufficient.

**Review by skhorasganiTT — APPROVED** [review]

**Review by eyonland — APPROVED** [review]

**Pull request merged**